### PR TITLE
De-sugar bang `!` operator - part 3 - optional `{} =`

### PR DESCRIPTION
**Pull request by lukewilliamboswell**
This PR adds to #6599 to make `{} =` optional for suffixed identifiers.

A statement is valid syntax if it beings with a suffixed identifier i.e. `Stdout.line!`. Multi-line statements will have an increased ident for the following lines. The `{} =` will be applied as sugar, and when formatted this will also be removed if it is present.

This PR was tested using the following programs (header excluded).

```roc
# Parses as three defs with a return expression 
main =
    a = "Foo"
    Stdout.line! a
    
    b = "Bar"
    Stdout.line b
```

```roc
# Parses as a single def with a return expression
main =
    Stdout.line! "Foo"

    Stdout.line "Bar"
```

### CODE REVIEW

File: crates/compiler/parse/src/ast.rs [hunk 1]
```diff
@@ -1135,6 +1138,8 @@ impl<'a> Pattern<'a> {
                     false
                 }
             }
+
+            Stmt(_) => todo!(),
```
Comment (rtfeldman) [inline crates/compiler/parse/src/ast.rs:1142]: Is it intentional to still have these two `todo!`s here, or should they actually be handled?

Comment (lukewilliamboswell) [reply crates/compiler/parse/src/ast.rs:1142]: I missed them. I think they cam be unreachable or internal_error. I'll have a closer look.

**Comment by lukewilliamboswell** [comment]
Fix for suffixed without arguments enables the following
```roc
main = run |> Task.onErr \err -> Stderr.line "ERROR: $(Inspect.toStr err)"

run =
    sayHello!
    checkJq!
    
    Task.ok {}

sayHello = Stdout.line "Hello World"

checkJq =
    "jq --version" 
    |> Cmd.new 
    |> Cmd.status
    |> Task.mapErr UnableToCheckJQVersion
```

**Comment by lukewilliamboswell** [comment]
Closing in favour of a new approach that I am working on. 

This was helpful to understand the scope of the task, but I think another direction will be more correct.

**Pull request closed without merging**